static/app/components/events/eventStatisticalDetector/eventDifferentialFlamegraph.tsx

```diff
@@ -54,20 +58,25 @@ export function EventDifferentialFlamegraph(props: EventDifferentialFlamegraphPr
     });
   }, [isValid, fingerprint, breakpoint]);
 
-  const projectID = parseInt(props.event.projectID, 10);
   const transactions = useTransactionsDelta();
+  const [transaction, setTransaction] = useState<
+    EventsResultsDataRow<string> | undefined
+  >(undefined);
+
+  if (transaction === undefined) {
+    const firstTransaction = transactions?.data?.data?.[0];
+    if (firstTransaction) {
+      setTransaction(firstTransaction);
+    }
+  }
```

Comment:
This part isn't very nice. We could potentially break it into 2 components where the outer component guarantees the transaction name is set before rendering the inner component to avoid this.

Reply:
Agree. I'll clean this up in a separate PR after I merge this last chain of changes